model/amity.py

```diff
@@ -0,0 +1,392 @@
+                raise ValueError(
+                    'name must be a string and not an integer or float'
+                    )
+            details = room.lower().split()
+            # ensure that atleast the room type and name is given
+            if len(details) > 2:
+                raise ValueError(
+                    'give just the type and name of the room or' +
+                    'livingspace without spaces'
+                    )
+            elif len(details) < 2:
+                raise ValueError('give at least room type and name')
+
+            name = [name for name in details if name not in (self.__room_type)]
+            name = ' '.join(name)
+            room_type = [typ for typ in details if typ in self.__room_type]
+            # ensure room type is only specified once
+            if len(room_type) > 1:
+                raise ValueError(
+                    'room type specifed {} times'.format(len(room_type))
+                    )
+            elif len(room_type) < 1:
+                raise ValueError(
+                    'room type not specified. should be office or livingspace'
+                    )
+            else:
+                room_type = ''.join(room_type)
+
+            if room_type == 'office':
+                if self.room_exists(name):
+                    raise ValueError(
+                        'room with name {} already exists.'.format(name)
+                        )
+                else:
+                    office = Office(name)
+                    # add attribute to hold occupants
+                    office.__dict__['occupants'] = []
+                    self.rooms.append(office.__dict__)
+                    self._results.append('office {} created'.format(name))
+
+            if room_type == 'livingspace':
+                if self.room_exists(name):
+                    raise ValueError(
+                        'room with name {} already exists.'.format(name)
+                        )
+                else:
+                    ls = LivingSpace(name)
+                    # add attribute to hold occupants
+                    ls.__dict__['occupants'] = []
+                    self.rooms.append(ls.__dict__)
+                    self._results.append('livingspace {} created'.format(name))
+        if len(self._results) > 1:
+            return self._results
+        else:
+            return self._results[0]
+
+    def allocate_space(self, person, livingSpace=False):
+        """This function allocates space to new employees."""
+        if person['type'].lower() == 'staff':
+            rooms = [
+                room for room in self.rooms
+                if room['room_type'] == 'office'
+                ]
+            try:
+                room = random.choice(rooms)
+            except IndexError:
+                raise IndexError('no rooms in amity to allocate person')
+            ind = self.rooms.index(room)
+            if self.check_room_availability(self.rooms[ind]['name']):
```

Comment:
here i wanted to check for the number of occupants in a room and if the room has space then append otherwise call the allocate method again since the allocations are random